### PR TITLE
Add Sharding sample, Java version

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

File: akka-sample-sharding-java/LICENSE [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Akka sample by Lightbend
+
+Licensed under Public Domain (CC0)
+
+To the extent possible under law, the person who associated CC0 with
+this Activator Tempate has waived all copyright and related or neighboring
+rights to this Activator Template.
```
Comment (patriknw) [inline akka-sample-sharding-java/LICENSE]: we should not use Activator here, haven't we changed that in other samples?
perhaps do a full sweep in another PR?

File: akka-sample-sharding-java/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+This tutorial contains a sample illustrating [Akka Cluster Sharding](http://doc.akka.io/docs/akka/current/scala/cluster-sharding.html#an-example).
```
Comment (patriknw) [inline akka-sample-sharding-java/README.md]: link to java docs

File: akka-sample-sharding-java/build.sbt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    scalaVersion := "2.12.2",
+    scalacOptions in Compile ++= Seq("-deprecation", "-feature", "-unchecked", "-Xlog-reflective-calls", "-Xlint"),
+    javacOptions in Compile ++= Seq("-Xlint:unchecked", "-Xlint:deprecation"),
+    javacOptions in doc in Compile := Seq("-Xdoclint:none"),
+    javaOptions in run ++= Seq("-Xms128m", "-Xmx1024m", "-Djava.library.path=./target/native"),
+    libraryDependencies ++= Seq(
+      "com.typesafe.akka" %% "akka-actor" % akkaVersion,
+      "com.typesafe.akka" %% "akka-remote" % akkaVersion,
+      "com.typesafe.akka" %% "akka-cluster" % akkaVersion,
+      "com.typesafe.akka" %% "akka-cluster-sharding" % akkaVersion,
```
Comment (patriknw) [inline akka-sample-sharding-java/build.sbt]: not really needed to list all dependencies, `akka-cluster-sharding` would be enough here

File: akka-sample-sharding-java/build.sbt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+val `akka-sample-sharding-java` = project
+  .in(file("."))
+  .settings(
+    organization := "com.typesafe.akka.samples",
+    scalaVersion := "2.12.2",
+    scalacOptions in Compile ++= Seq("-deprecation", "-feature", "-unchecked", "-Xlog-reflective-calls", "-Xlint"),
+    javacOptions in Compile ++= Seq("-Xlint:unchecked", "-Xlint:deprecation"),
+    javacOptions in doc in Compile := Seq("-Xdoclint:none"),
+    javaOptions in run ++= Seq("-Xms128m", "-Xmx1024m", "-Djava.library.path=./target/native"),
```
Comment (patriknw) [inline akka-sample-sharding-java/build.sbt]: the library.path was for Sigar metrics, not used here

File: akka-sample-sharding-java/src/main/java/sample/sharding/Device.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+public class Device extends AbstractActor {
+
+  private LoggingAdapter log = Logging.getLogger(getContext().system(), this);
+
+  private List<Double> temperatures = new ArrayList<Double>();
+
+  public static class RecordTemperature implements Serializable {
+    final Integer deviceId;
+    final Double temperature;
```
Comment (patriknw) [inline akka-sample-sharding-java/src/main/java/sample/sharding/Device.java]: make these `public`

File: akka-sample-sharding-java/src/main/java/sample/sharding/Devices.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+  private final LoggingAdapter log = Logging.getLogger(getContext().system(), this);
+
+  private final ActorRef deviceRegion;
+
+  private final Random random = new Random();
+
+  private final Integer numberOfDevices = 50;
+
+  ShardRegion.MessageExtractor messageExtractor = new ShardRegion.MessageExtractor() {
```
Comment (patriknw) [inline akka-sample-sharding-java/src/main/java/sample/sharding/Devices.java]: `static`

File: akka-sample-sharding-java/src/main/resources/application.conf [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  cluster {
+    seed-nodes = [
+      "akka.tcp://ShardingSystem@127.0.0.1:2551",
+      "akka.tcp://ShardingSystem@127.0.0.1:2552"]
+
+    # auto downing is NOT safe for production deployments.
+    # you may want to use it during development, read more about it in the docs.
+    auto-down-unreachable-after = 10s
+  }
+}
```
Comment (patriknw) [inline akka-sample-sharding-java/src/main/resources/application.conf]: line feed in conf files

File: akka-sample-sharding-java/LICENSE (hunk 1 above)
Comment (raboof) [inline akka-sample-sharding-java/LICENSE]: Apparently not (as this was of course copy-pasta :) ) - added #33

File: akka-sample-sharding-java/src/main/resources/application.conf (hunk 7 above)
Comment (raboof) [inline akka-sample-sharding-java/src/main/resources/application.conf]: It'd be really neat if we could have sbt take care of this instead of relying on our editors.
